modules/fx/arrow-fx/src/main/kotlin/arrow/fx/IO.kt

```diff
@@ -461,3 +428,36 @@ fun <B, E : B, A> IOOf<E, A>.widenError(): IO<B, A> =
   fix()
 
 fun <A> A.liftIO(): IO<Nothing, A> = IO.just(this)
+
+/**
+ * Create a new [IO] that upon execution starts the receiver [IO] within a [Fiber] on [ctx].
+ *
+ * ```kotlin:ank:playground
+ * import arrow.fx.*
+ * import arrow.fx.extensions.fx
+ * import kotlinx.coroutines.Dispatchers
+ *
+ * fun main(args: Array<String>) {
+ *   //sampleStart
+ *   val result = IO.fx {
+ *     val (join, cancel) = !IO.effect {
+ *       println("Hello from a fiber on ${Thread.currentThread().name}")
+ *     }.startFiber(Dispatchers.Default)
+ *   }
+ *
+ *   //sampleEnd
+ *   result.unsafeRunSync()
+ * }
+ * ```
+ *
+ * @receiver [IO] to execute on [ctx] within a new suspended [IO].
+ * @param ctx [CoroutineContext] to execute the source [IO] on.
+ * @return [IO] with suspended execution of source [IO] on context [ctx].
+ */
+fun <A> IO<Throwable, A>.startFiber(ctx: CoroutineContext): IO<Throwable, Fiber<IOPartialOf<Throwable>, A>> = IO.async { cb ->
```

Comment:
This can be parameterized to `E` if you add a `f: (Throwable) -> E`, and then you can provide a default override where `E` is `Throwable`.

`fun <A> IO<Throwable, A>.startFiber(ctx: CoroutineContext): IO<Throwable, Fiber<IOPartialOf<Throwable>, A>>`